activesupport/lib/active_support/number_helper/number_to_rounded_converter.rb

```diff
@@ -30,7 +30,7 @@ def convert
           formatted_string =
             if BigDecimal === rounded_number && rounded_number.finite?
               s = rounded_number.to_s('F') + '0'*precision
-              a, b = s.split('.', 2)
+              a, b = s.split('.'.freeze, 2)
               a + '.' + b[0, precision]
```

Comment:
We could further reduce allocations by appending to the "a" string here using << instead of +. We can do that because split returns a new string. I think we can use the same trick above where we are adding the zeroes to generate "s".